sympy/matrices/common.py

```diff
@@ -2205,6 +2205,113 @@ def __sub__(self, a):
     def __truediv__(self, other):
         return self.__div__(other)
 
+    def _eval_matrix_mul_strassen(self, other):
+        """Strassen subroutine for matrix multiplication.
+
+        References
+        ==========
+
+        .. [1] https://en.wikipedia.org/wiki/Strassen_algorithm
+        """
+        a_rows, a_cols = sympify(self.shape)
+        b_rows, b_cols = sympify(other.shape)
+
+        def _ceiling_2(i):
+            """Helper function to decide augmented matrix size, which
+            should have even dimensions.
+            """
+            if i.is_odd:
+                return i + 1
+            else:
+                return i
+
+        def _pad_zeros(mat, i_old, j_old, i_new, j_new):
+            """Helper function to pad zeros to the matrix
+            """
+            cls = mat.__class__
+            if i_new > i_old:
+                diff = i_new - i_old
+                mat = cls.vstack(mat, cls.zeros(diff, j_old))
+            if j_new > j_old:
+                diff = j_new - j_old
+                mat = cls.hstack(mat, cls.zeros(i_new, diff))
+            return mat
+
+        _strasson = MatrixArithmetic._eval_matrix_mul_strassen
+
+        if (a_rows, a_cols) == (1, 1) and (b_rows, b_cols) == (1, 1):
```

Comment:
The Strassen algorithm wouldn't normally be used all the way down to 1x1. Maybe there should be an argument to specify the minimum size while recursing.